docs/docs/development/ide-integration.md

```diff
@@ -0,0 +1,139 @@
+import TabItem from '@theme/TabItem';
+
+export const OsTabs = (props) => (<Tabs
+groupId="operating-systems"
+defaultValue="debian"
+values={[
+{label: 'Debian/Ubuntu', value: 'debian'},
+{label: 'Windows', value: 'win'},
+{label: 'macOS', value: 'mac'},
+{label: 'Raspberry OS', value: 'raspberryos'},
+{label: 'Fedora', value: 'fedora'},
+{label: 'VS Code & Docker', value: 'docker'},
+]
+}>{props.children}</Tabs>);
+
+## Visual Studio Code
+
+Visual Studio Code needs to know some things about the project such as include
+paths and compiler paths before features such as code completion, go to definition,
+and graying out disabled code blocks will work. Fortunately, CMake can generate
+that configuration for us automatically.
+
+### Create a Compilation Database
+
+To configure `west` to tell CMake to generate a compilation database, open a
+terminal to the ZMK repository and run the following command:
+
+```sh
+west config build.cmake-args -- -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+```
+
+Every [build](build-flash#building) will now update the database. You will
+need to build once to create the database before code completion will work.
+We'll tell Visual Studio Code where to find the database in the next step.
+
+:::note
+If you have set any other CMake arguments such as the path to your zmk-config, the
+above command will overwrite them. You should instead provide the flag to export
+compile commands and all other arguments surrounded by quotes. For example:
+
+```sh
+west config build.cmake-args -- "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DZMK_CONFIG=/path/to/zmk-config/config"
+```
+
+:::
+
+### Create a C/C++ Configuration
+
+Install the [C/C++ extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools),
+then run **F1 > C/C++: Edit Configurations (UI)**. It should automatically create
+a new configuration for you, but if the text box under **Configuration name** is empty,
+click **Add Configuration**, enter a name, and click **OK**.
+
+Change these options:
+
+| Option                                | Value                                                |
+| ------------------------------------- | ---------------------------------------------------- |
+| Compiler path                         | Path to your toolchain's GCC binary (see below)      |
+| IntelliSense mode                     | gcc-arm                                              |
+| Advanced Settings > Compiler commands | `${workspaceFolder}/app/build/compile_commands.json` |
+
+<OsTabs>
+<TabItem value="debian">
+
+Open VS Code's integrated terminal and run the following commands. It will print
+your compiler path.
+
+```sh
+source zephyr/zephyr-env.sh
+echo ${ZEPHYR_SDK_INSTALL_DIR}/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
+```
+
+:::note
+You will need to update this path any time you switch to a new version of the Zephyr SDK.
+:::
+
+</TabItem>
+<TabItem value="win">
+
+Your compiler path is
+
+```
+${env:GNUARMEMB_TOOLCHAIN_PATH}/bin/arm-none-eabi-gcc.exe
```

Comment:
I didn't realize that some of these tabs show an echo command that you copy the result of and some you just copy and paste (no need to run in cmd/bash). Not sure if the could be made consistent maybe? Or maybe a bit more obvious for people who are quickly skimming 😅

Reply:
I don't think we can make it be consistent because for some configurations (Docker, Windows), VS Code is running with the variables pointing to the compiler path already in its environment, but for others you have to source Zephyr first so only the integrated terminal will see those variables in its environment.

Reply:
I suppose we could convert them all to echo commands, but then we're not taking advantage of the environment variables for platforms that support it. I'm not sure why I didn't realize this at first, so not sure what wording would make it easier to understand. This is probably fine then.